torch/lib/THC/THCTensorMathPointwise.cuh

```diff
@@ -415,11 +415,17 @@ struct TensorDivOp<half> {
 template <typename T>
 struct TensorCRemainderOp {
   __device__ __forceinline__ void operator()(T* out, T* in) {
-    *out = *in != 0 ? *out - *in * (*out / *in) : NAN;
+    *out = (*in != 0) ? *out % *in : NAN;
+    if ((*out * *in)<0){
+      *out += *in;
+    }
   }
 
   __device__ __forceinline__ void operator()(T* out, T* in1, T* in2) {
-    *out = *in2 != 0 ? *in1 - *in2 * (*in1 / *in2) : NAN;
+    *out = (*in2 != 0) ?  *in1 % *in2 : NAN;
```

Comment:
I'm confused here, is it an implementation for integers? `%` is only defined for integral types, but there's no `NAN` for them. I think we should just use `*in1 % *in2`. BTW what happens when a GPU divides by 0? Does it raise a CUDA error?